aidants_connect_web/models.py

```diff
@@ -274,14 +275,30 @@ class Mandat(models.Model):
 
     objects = MandatQuerySet.as_manager()
 
+    def __str__(self):
+        return f"#{self.id}"
+
     @property
     def is_expired(self) -> bool:
         return timezone.now() > self.expiration_date
 
+    @cached_property
+    def is_active(self):
```

Comment:
je suis un peu perdu si c'est l'autorisation ou le mandat qui porte l'information du "is_active". En gros c'est bien une combinaison des 2 : si le mandat n'est pas encore expiré + si l'autorisation n'est pas révoquée. Parce qu'entre le queryset du Mandat, cette property, et le queryset de l'Autorisation et les propriétés de l'Autorisation on se répète je trouve, y'aurait peut-etre moyen de keep it DRY ?

Reply:
Dans l'absolu, je me dis qu'on peut vouloir répondre à la question : "ce _truc_ est-il actif ?" aussi bien du point de vue d'une autorisation que d'un mandat, du coup ça ne me parait pas déconnant d'avoir une méthode ou un attribut `is_active` dans chacune des deux classes.

Si j'exprime les règles en français :
- Un `mandat` est actif tant qu'il contient au moins une `autorisation` active
- Une `autorisation` est active si elle n'a pas été révoquée ET si la date d'expiration de son `mandat` n'a pas été atteinte

On pourrait effectivement légèrement optimiser le check du `mandat` car il _suffit_ que sa date d'expiration soit atteinte pour qu'il soit inactif, sans qu'il soit nécessaire de vérifier toutes ses `autorisations`.

Après, les "répétitions" en sont-elles vraiment, c'est discutable :)
À mon sens l'objectif est surtout de fournir une API riche qui permette ensuite d'utiliser facilement les objets, et d'éviter les répétitions _dans le code appelant_. Par exemple, ici, à partir de la classe, je peux facilement récupérer les instances actives (grâce au queryset) et/ou savoir immédiatement, lisiblement, si une instance donnée est active ou non (`truc.is_active`), sans avoir besoin de me replonger dans les détails de l'implémentation, qui sont complètement "contenus" dans le modèle. De fait, on arrive dans des considérations plus "archi" en quelque sorte — à ta dispo pour en discuter davantage, en particulier de visu cet après-midi ^^